### PR TITLE
SSCS-10273 Call LOV reference data for HMC call

### PR DESCRIPTION
**Pull request by bensonlievoco**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-10273

### Change description ###
- Created serval enum base on LOV data.
- Extended codestyle line length check to 240

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/model/lov/ref/data/CancellationReason.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    private final String key;
+    private final String value;
+
+    CancellationReason(String key, String value) {
+        this.key = key;
+        this.value = value;
+    }
+
+    public static CancellationReason getCancellationReasonByValue(String value) {
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/lov/ref/data/CancellationReason.java:26]: This flow is a bit old school.. we don't need to assign to a variable, can return value within the for loop or return null if no matches.

File: src/main/java/uk/gov/hmcts/reform/sscs/model/lov/ref/data/CaseSubType.java [hunk 2]
```diff
@@ -0,0 +1,1385 @@
...
+    private final CaseType caseType;
+
+    CaseSubType(String key, String value, CaseType caseType) {
+        this.key = key;
+        this.value = value;
+        this.caseType = caseType;
+    }
+
+
+    public static CaseSubType getCaseSubTypeBySubTypeValueAndCaseType(String value, CaseType caseType) {
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/lov/ref/data/CaseSubType.java:1360]: This method name is a bit confusing change to.. `getCaseSubTypeByValueAndCaseType` ? or I'd prefer just `getCaseSubType` the parameters tell you what you need to pass in.

File: src/main/java/uk/gov/hmcts/reform/sscs/model/lov/ref/data/CaseType.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
+package uk.gov.hmcts.reform.sscs.model.lov.ref.data;
+
+public enum CaseType {
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/lov/ref/data/CaseType.java:3]: can you remove new lines between properties in this enum?

**Review by wmarshid — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/model/lov/ref/data/CancellationReason.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.reform.sscs.model.lov.ref.data;
+
+public enum CancellationReason {
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/lov/ref/data/CancellationReason.java:3]: The enums should be in sscs-common

**Pull request closed without merging**